test/regression.decorator-this.test.js

```diff
@@ -0,0 +1,47 @@
+const Code = require('@hapi/code')
+const Lab = require('@hapi/lab')
+const Util = require('util')
+
+const lab = (exports.lab = Lab.script())
+const describe = lab.describe
+const expect = Code.expect
+const it = lab.it
+
+const Seneca = require('..')
+
+
+describe('regression test, this-ref in a decorator', () => {
+  it('calls the decorator with the correct this-ref', test((fin) => {
+    const si = Seneca()
+    si.test(fin)
+
+
+    si.decorate('amI', function (self) {
+      return this === self
+    })
+
+
+    si.add('hello:world', function (args, reply) {
+      const delegate = this.delegate()
+      const result = delegate.amI(delegate)
+      expect(result).equal(true)
```

Comment:
This fails because functions can only be bound once, binding multiple times does not affect the this-ref:
https://github.com/senecajs/seneca/blob/611c193348e464a961fc2077fbc38acd16c1d798/lib/api.ts#L309